packages/ses/scripts/bundle.js

```diff
@@ -16,40 +20,60 @@ const write = async (target, content) => {
   await fs.promises.writeFile(location, content);
 };
 
-const main = async () => {
+/**
+ * @param {object} [options]
+ * @param {string} [options.buildType] Suffix used to build special bundles (e.g. 'hermes')
+ */
+const writeBundle = async ({ buildType } = {}) => {
   const text = await fs.promises.readFile(
     fileURLToPath(`${root}/package.json`),
     'utf8',
   );
   const packageJson = JSON.parse(text);
   const version = packageJson.version;
 
+  /** @type ModuleTransforms */
+  const moduleTransforms = {};
+
+  let bundleFilePaths = [
+    'dist/ses.cjs',
+    'dist/ses.mjs',
+    'dist/ses.umd.js',
+    'dist/lockdown.cjs',
+    'dist/lockdown.mjs',
+    'dist/lockdown.umd.js',
+  ];
+  let terseFilePaths = ['dist/ses.umd.min.js', 'dist/lockdown.umd.min.js'];
+
+  if (buildType === 'hermes') {
+    bundleFilePaths = ['dist/ses-hermes.cjs'];
+    terseFilePaths = [];
+    Object.assign(moduleTransforms, hermesTransforms);
+  }
+
   const bundle = await makeBundle(
     read,
     pathToFileURL(resolve('../index.js', import.meta.url)).toString(),
+    { moduleTransforms },
```

Comment:
`makeBundle` accepts a `conditions` option, which can be `new Set(['hermes'])` to use a `"hermes"` package condition in `package.json`. My hope is that you can simply `import "ses"` in Hermes applications instead of `import "ses/hermes.js"`.

Regarding the `".js"` extension in my causal misspelling of `import "ses/hermes"`, we’ve been burned too many times by tools that do not yet pay attention to `"exports"` in `package.json`, so have a general rule that the public exports should be shadowed by a physical file with the `".js"` extension, like `ses/lockdown-shim.js`. I can imagine beginning to relax that rule for any code that is provably unusable without `"exports"` support, but I want to frame that as a rule and not an exception before encouraging its use.

Reply:
> `makeBundle` accepts a `conditions` option, which can be `new Set(['hermes'])` to use a `"hermes"` package condition in `package.json`. My hope is that you can simply `import "ses"` in Hermes applications instead of `import "ses/hermes.js"`.

it's great the `conditions` option exists, but i'm unsure how a Hermes app would `import "ses"` and somehow resolve this to `dist/ses-hermes.cjs`, perhaps i don't fully understand how the `conditions` option works

Reply:
> Regarding the `".js"` extension in my causal misspelling of `import "ses/hermes"`, we’ve been burned too many times by tools that do not yet pay attention to `"exports"` in `package.json`, so have a general rule that the public exports should be shadowed by a physical file with the `".js"` extension, like `ses/lockdown-shim.js`. I can imagine beginning to relax that rule for any code that is provably unusable without `"exports"` support, but I want to frame that as a rule and not an exception before encouraging its use.

i can see `"./hermes.js": "./dist/ses-hermes.cjs",` working well

so in `"./hermes.js": "./hermes.js",` would the physical extra shadow file be a copy of `dist/ses-hermes.cjs`?